project2/vehicleComms.cpp

```diff
@@ -0,0 +1,50 @@
+#include <stdint.h>
+#include <Arduino.h>
+#include "vehicleComms.h"
+#include "schedule.h"
+
+void vehicleComms(void *vehicleCommsData) {
+
+    // Don't run if it has been less than a major cycle since last execution
+    static unsigned long lastRunTime;
+    if (globalTimeBase() - lastRunTime < MAJOR_CYCLE_DURATION_MS) {
+        return;
+    }
+    lastRunTime = globalTimeBase();
+
+    // Cast from void to correct type
+    VehicleCommsData *data = (VehicleCommsData *) vehicleCommsData;
+
+    char comm = *data->command;
+    
+    // comm = 'F';
+    // Print command to Serial1 for the Uno to pick up
+    Serial1.println(comm);
+
+    char response;
+    // TODO: This it probably not a good way to do an interupt, shouldn't use MILLIS
+    unsigned long time = millis();
```

Comment:
Maybe this is what they meant by the dynamically scheduled serial task.

Also, instead of waiting for a response, I think it should skip writing for now until there is data available. Check for new data whether or not there was a command sent during this task. That way you don't have to wait and can let the next run handle it.